src/code/z_demo.c

```diff
@@ -235,96 +247,118 @@ void Cutscene_Command_Misc(PlayState* play, CutsceneContext* csCtx, CsCmdBase* c
                 }
             }
             break;
-        case 0xE:
+
+        case CS_MISC_HALT_ALL_ACTORS:
             play->haltAllActors = true;
             break;
-        case 0xF:
+
+        case CS_MISC_RESUME_ALL_ACTORS:
             play->haltAllActors = false;
             break;
-        case 0x10:
-            if (isStartFrame) {
+
+        case CS_MISC_SANDSTORM_FILL:
+            if (isFirstFrame) {
                 play->envCtx.sandstormState = 1;
             }
             func_8019F128(NA_SE_EV_SAND_STORM - SFX_FLAG);
             break;
-        case 0x11:
+
+        case CS_MISC_SUNSSONG_START:
             gSaveContext.sunsSongState = SUNSSONG_START;
             break;
-        case 0x12:
+
+        case CS_MISC_FREEZE_TIME:
             if (!gSaveContext.save.isNight) {
                 gSaveContext.save.time = ((void)0, gSaveContext.save.time) - (u16)R_TIME_SPEED;
             } else {
                 gSaveContext.save.time = ((void)0, gSaveContext.save.time) - (u16)(2 * R_TIME_SPEED);
             }
             break;
-        case 0x13:
-            AudioOcarina_PlayLongScarecrowAfterCredits();
-            csCtx->frames = cmd->startFrame - 1;
+
+        case CS_MISC_LONG_SCARECROW_SONG:
+            AudioOcarina_PlayLongScarecrowSong();
+            csCtx->curFrame = cmd->startFrame - 1; // the cutscene runs forever
             break;
-        case 0x14:
-            EnvFlags_Set(play, 3);
+
+        case CS_MISC_SET_CSFLAG_3:
+            CutsceneFlags_Set(play, 3);
             break;
-        case 0x15:
-            EnvFlags_Set(play, 4);
+
+        case CS_MISC_SET_CSFLAG_4:
+            CutsceneFlags_Set(play, 4);
             break;
-        case 0x16:
+
+        case CS_MISC_PLAYER_FORM_DEKU:
             gSaveContext.save.playerForm = PLAYER_FORM_DEKU;
             break;
-        case 0x17:
+
+        case CS_MISC_ENABLE_PLAYER_REFLECTION:
             player->stateFlags2 |= PLAYER_STATE2_4000000;
             break;
-        case 0x18:
+
+        case CS_MISC_DISABLE_PLAYER_REFLECTION:
             player->stateFlags2 &= ~PLAYER_STATE2_4000000;
             break;
-        case 0x19:
+
+        case CS_MISC_PLAYER_FORM_HUMAN:
             sCutsceneStoredPlayerForm = gSaveContext.save.playerForm;
             gSaveContext.save.playerForm = PLAYER_FORM_HUMAN;
             gSaveContext.save.equippedMask = PLAYER_MASK_NONE;
             break;
-        case 0x1A:
+
+        case CS_MISC_STRONG_QUAKE_START:
             func_8019F128(NA_SE_EV_EARTHQUAKE_LAST2 - SFX_FLAG);
-            if (isStartFrame) {
+            if (isFirstFrame) {
```

Comment:
same here i suppose. 
also the command has "start" but its playing a sound. i assume if you want the sound to play you need the command to last the full duration, cause of the SFX_FLAG. so the "start" in the name might be misleading

Reply:
looking again at the sfx names, it seems like a pretty specific sequence of quake sounds. is this *only* used in like cutscenes where the moon is falling? it may be worth being way more specific with the name in that case

Reply:
Good catch about the `SFX_FLAG` and needing the cmd to be sustained between `startFrame` and `endFrame`. Will definitely drop `START`.

I think:
`CS_MISC_EARTHQUAKE_STRONG`
`CS_MISC_EARTHQUAKE_MEDIUM`
`CS_MISC_EARTHQUAKE_WEAK`
should be thought of as a group of misc cmds, and are better differentiated by the strength of the quake, since that is the key parameter differentiating them. Seems like it could easily be general purpose, so I don't think it needs to be tied to moon crash, even if it is only used there.

But I will listen to `NA_SE_EV_EARTHQUAKE_LAST` vs `NA_SE_EV_EARTHQUAKE_LAST2` to see if those sound different enough

Reply:
I am wondering about `CS_MISC_QUAKE_STOP` though, I ownder if that's worth renaming to `CS_MISC_EARTHQUAKE_STOP`. Also note that cmd calls `Quake_Init`, which is just poorly named and should more be `Quake_Reset`

Reply:
it is kinda weird that the normal quake commands from oot are gone. 
technically you can call the command for 1 frame and it will queue up the quake and only play the sound for a frame which may be barely audible?

itd be hella jank, but it could be how they do a generic quake outside of moon crash scenes. unless there are just no quakes in other contexts. 
looking at usages in extracted cutscenes will give helpful context

Reply:
`CS_MISC_EARTHQUAKE_WEAK` has 3 uses in Ikana Valley
`CS_MISC_EARTHQUAKE_MEDIUM` has 5 uses in Termina Field, 3 uses in Woodfall, 1 use in Clock Tower Rooftop
`CS_MISC_EARTHQUAKE_STRONG` has 1 use in Termina Field

Reply:
ok, alot of these are moon-related but not all of them. ikana is probably the music box house stuff, and woodfall is the dungeon appearing

I think it makes sense to call this quake + sound combination "earthquake" and since there is no standalone quake command anymore, the stop command can also be named after earthquake

Reply:
tho it is confusing, cause to stop the sound, you need to stop the other command. but to stop the quake you need to run the stop quake command 
NotLikeThis

Reply:
its a really dumb situation. calling it "quake stop" makes it seem like there should be a generic "quake" command like oot has. 
but calling it "earthquake stop" only stops the quake part of the earth quake, not the sound. so its not fully fair to tie it to the "earthquake" concept

I lean toward naming it after just quake, but not very sure

Reply:
Yeah, that's tricky. I named it `CS_MISC_EARTHQUAKE_STOP` because it does parallel nicely, but as you mentioned it only stops half of what `CS_MISC_EARTHQUAKE_XXX` is about, so maybe `CS_MISC_QUAKE_STOP` is better